File: service/router/panel/itemPage.go
package panel

import (
	"sun-panel/api/api_v1"
	"sun-panel/api/api_v1/middleware"

	"github.com/gin-gonic/gin"
)

func InitItemPage(router *gin.RouterGroup) {
	itemPage := api_v1.ApiGroupApp.ApiPanel.ItemPage
	r := router.Group("", middleware.LoginInterceptor)
	{
		r.POST("/panel/itemPage/edit", itemPage.Edit)
		r.POST("/panel/itemPage/deletes", itemPage.Deletes)
		r.POST("/panel/itemPage/saveSort", itemPage.SaveSort)
	}

	// 公开模式
	rPublic := router.Group("", middleware.PublicModeInterceptor)
	{
		rPublic.POST("/panel/itemPage/getList", itemPage.GetList)
	}
}

File: src/api/panel/itemPage.ts
import { post } from '@/utils/request'

// 获取页面列表
export function getList<T>() {
  return post<T>({ url: '/panel/itemPage/getList', data: {} })
}

// 编辑页面
export function edit<T>(data: any) {
  return post<T>({ url: '/panel/itemPage/edit', data })
}

// 删除页面
export function deletes<T>(ids: number[]) {
  return post<T>({ url: '/panel/itemPage/deletes', data: { ids } })
}

// 保存页面排序
export function saveSort<T>(sortItems: Array<{ id: number; sort: number }>) {
  return post<T>({ url: '/panel/itemPage/saveSort', data: { sortItems } })
}
